### PR TITLE
Add support for canonical URLs

### LINKED ISSUE #844: canonical links should point to latest or stable
Historical documentation (hammer and below) currently point to themselves with [canonical links](https://en.wikipedia.org/wiki/Canonical_link_element), but instead should point to stable or latest.  The reason this is preferable is because when people google for documentation we _don't_ want them to find results for old versions and become confused.  Instead, google should only return the latest documentation, and users who need older documentation can click through to older versions.

### PR DESCRIPTION
**Pull request by Fryguy**
Fixes #844

@chessbyte Please review.

### CODE REVIEW

**Comment by Fryguy** [comment]
Opened issue #866 for the API docs not having versions.

**Comment by Fryguy** [comment]
Just noticed this may have broken the sorting of the blog posts (somehow)

**Comment by Fryguy** [comment]
@chessbyte This is ready to go.

File: site/_plugins/canonical_url.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Jekyll::Page#url is memoized on first call, but this causes issues with the
+#   paginator.  As such, we get the URL and then remove the memoization.
+def extract_page_url(page)
+  had_url = true if page.instance_variable_defined?(:@url)
```
Comment (chessbyte) [inline site/_plugins/canonical_url.rb:4]: why not the simpler `had_url = page.instance_variable_defined?(:@url)` ?

Comment (Fryguy) [reply site/_plugins/canonical_url.rb:4]: Oh yeah, that's better...I had that originally, but changed things, and somehow ended with this.

**Comment by Fryguy** [comment]
@chessbyte Updated.

**Pull request merged**